modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -664,16 +816,89 @@ public void testPasswordHistoryWithModifiedEncryption() throws Exception {
 					portalException.getClass());
 
 				Assert.assertEquals(
-					"{SHA-384}66b6aa56af08dc8caf7e001683058338244f436de61d40" +
-						"e342d0c69bda9f73cd6d167fdb29925db579923bdcef1fe5ae",
-					user.getPassword());
+					Authenticator.SUCCESS,
+					_userLocalService.authenticateByEmailAddress(
+						user.getCompanyId(), user.getEmailAddress(), password2,
+						null, null, null));
 			}
 			finally {
 				ServiceContextThreadLocal.popServiceContext();
 			}
 		}
 	}
 
+	@Test
+	public void testPasswordHistoryWithModifiedEncryption() throws Exception {
+		try (AutoCloseable autoCloseable1 =
+				ReflectionTestUtil.setFieldValueWithAutoCloseable(
+					DigesterImpl.class, "_BASE_64", false);
+			AutoCloseable autoCloseable2 =
+				ReflectionTestUtil.setFieldValueWithAutoCloseable(
+					PasswordEncryptorUtil.class,
+					"_PASSWORDS_ENCRYPTION_ALGORITHM", "SHA-384")) {
+
+			try (SafeCloseable safeCloseable =
+					_updateDefaultPasswordPolicyWithSafeCloseable(
+						passwordPolicy -> {
+							passwordPolicy.setHistory(true);
+							passwordPolicy.setHistoryCount(2);
+						})) {
+
+				String password1 = "password1";
+				String password2 = "password2";
+
+				User user = UserTestUtil.addUser();
```

Comment:
Here you've declared these variables all together and in a different order than in `testPasswordHistory` above. You need to keep declarations consistent, both in relative position and order.